shlr/gdb/src/arch.c

```diff
@@ -76,27 +77,27 @@ gdb_reg_t *arch_parse_reg_profile(const char *reg_profile) {
 			if (*p == '#') {
 				// Place the rest of the line in the token if a comment is encountered
 				for (i = 0; *p != '\n'; p++) {
-					if (i < sizeof(tmp) - 1) {
+					if (i < BUFFER_SIZE - 1) {
 						tmp[i++] = *p;
 					}
 				}
 			} else {
 				// Save all characters up to a space/tab
 				// Use isgraph instead of isprint because the latter considers ' ' printable
-				for (i = 0; isgraph((const unsigned char)*p) && i < sizeof(tmp) - 1;) {
+				for (i = 0; isgraph((const unsigned char)*p) && i < BUFFER_SIZE - 1;) {
 					tmp[i++] = *p++;
 				}
 			}
 			tmp[i] = '\0';
 			// Limit the number of tokens
-			if (j > PARSER_MAX_TOKENS - 1) {
+			if (j >= PARSER_MAX_TOKENS) {
 				break;
 			}
 			// Save the token
 			tok[j++] = strdup(tmp);
 		}
 		// Empty line, eww
-		if (j) {
+		if (j == 4) {
```

Comment:
@yossizap could you have a look and approve if you feel it is ok?

Reply:
I don't think that this is right, 5 is the maximum number of tokens here but it can be higher(defined as 8 in profile.c).

This entire function is a duplication of `rz_reg_set_profile_string` from `librz/reg/profile.c` since that API isn't accessible here and RzReg isn't available(this is bad and hopefully will change soon). The implementation in `profile.c` also properly treats lines that start with `=` that are j==2 and there must be other edge cases here with flag tokens etc. I would keep it the way it was until we move this out of `shlr`(this is only reached if the xml protocol fails and the default regprofiles available here aren't really complex atm) or try to adopt `profile.c`'s implementation temporarily.

Reply:
I will revert this commit. Thanks